### DISCUSSION TITLE
How does conform form dirty status  turn to be false after submission success by Nextjs server action??

### DISCUSSION THREAD

**Discussion by takakikasuga** [edmundhung/conform#819] [Q&A]
@edmundhung Hi!!

## My Hope

After form submission, I want the latest submitted form values to become the default values, and the form's dirty status to reset to false. Additionally, the submit button should remain disabled unless the user enters a new value after the submission.

In React Hook Form, [the reset method](https://react-hook-form.com/docs/useform/reset) is available for this purpose.
Is there a similar reset method in the Conform library?

### conform settings

```typescript
  const [lastResult, action] = ReactDom.useFormState(edit, undefined);
  const [form, fields] = useForm<EditSchema>({
    defaultValue: {
      description,
      instagram,
      name,
      tiktok,
      twitter,
      username,
      youtube,
    },
    lastResult,
    onValidate: ({ formData }) => parseWithZod(formData, { schema: editSchema }),
    shouldRevalidate: "onInput",
    shouldValidate: "onBlur",
  });
```

### Server Actions

```typescript

export const edit = async (prevState: unknown, formData: FormData) => {
  const session = await auth();
  const { env } = getRequestContext();

  if (!session) {
    redirect("/");
  }

  const submission = parseWithZod(formData, {
    schema: editSchema,
  });

  if (submission.status !== "success") {
    return submission.reply();
  }

  await patchApiV1CreatorsMe(
    {
      description: submission.value.description,
      name: submission.value.name,
      socialMediaAccounts: {
        instagram: submission.value.instagram,
        tiktok: submission.value.tiktok,
        twitter: submission.value.twitter,
        youtube: submission.value.youtube,
      },
      username: submission.value.username,
    },
    {
      headers: {
        Authorization: `Bearer ${session.user.auth.accessToken}`,
      },
    }
  );

  return submission.reply();
};
```

thk!!

**Comment by safinn** [comment]
You can return `return submission.reply({ resetForm: true })` from your action.
I want to do the same in react router v7 but with `{ resetForm: true }` my form is no longer dirty but the default value is not right. Without reset, the default value changes correctly after update but the dirty status does not reset.

**Reply by mephju** [reply]
@edmundhung 
I'm experiencing the same issue as @safinn. Is this the intended behavior?

**Reply by mephju** [reply]
Ah ok. 
I saw Edmund's tip in the Remix section.
@safinn You might want to read it:
https://conform.guide/integration/remix#tips
